### PR TITLE
upcoming: [M3-8340] - Update Manage Image Regions Drawer based on UX feedback 

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Updates the "Manage Image Regions" Drawer based on UX feedback 🖌️ 

## Changes  🔄
- The Manage Regions Multi-Select should not show any tags in the TextField 🚫🏷️
- Regions should not be filtered from the Multi-Select 🌐 
- Make last region immutable when there is only one region left in the list ✖️ 
- Close the drawer when "Save" is pressed (and the API request is successful 📕 

## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115251059/5e5ec345-8a1e-40a4-9c42-de0892728a84" /> | <video src="https://github.com/linode/manager/assets/115251059/d3180a75-3bc8-4c15-9115-f594f227f498" /> |

### Prerequisites
- Turn on Image Service Gen 2 feature flag
- Turn on the MSW

### Verification steps
- Test the "Manage Regions" feature on the  Image called `multi-regions-test-image` 🧪 
  - Verify all of the changes listed above are implemented

## As an Author I have considered 🤔

- [ ] 👀 Doing a self review
- [ ] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/components/RegionSelect/RegionMultiSelect.tsx [hunk 1]
```diff
@@ -155,7 +155,6 @@ export const RegionMultiSelect = React.memo((props: RegionMultiSelectProps) => {
             InputProps: {
               required,
             },
-            placeholder: selectedRegions.length > 0 ? '' : placeholder,
```
Comment (bnussman-akamai) [inline packages/manager/src/components/RegionSelect/RegionMultiSelect.tsx:158]: I don't think this is a good default so I moved this special logic into `AccessKeyRegions.tsx`. 

We want the placeholder to always show in the Manage Regions Drawer and this logic was messing that up

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyRegions/AccessKeyRegions.tsx [hunk 2]
```diff
@@ -29,13 +29,15 @@ export const AccessKeyRegions = (props: Props) => {
 
   return (
     <RegionMultiSelect
-      onChange={onChange}
+      placeholder={
+        selectedRegion.length > 0 ? ' ' : 'Select regions or type to search'
```
Comment (hkhalil-akamai) [inline packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyRegions/AccessKeyRegions.tsx:33]: Is there a reason for this space?

```suggestion
        selectedRegion.length > 0 ? '' : 'Select regions or type to search'
```

**Review by hkhalil-akamai — APPROVED** [review]
I'm strongly liking this new iteration of the manage regions flow and I think it improves on a lot of the usability quirks of the original version.

Verified all the changes and the existing functionality continue to work as expected. Also verified `RegionMultiSelect` works as expected in `AccessKeyRegions`.

Thanks for the cleanup in `RegionMultiSelect` and for test coverage.

File: packages/manager/src/features/Images/ImagesLanding/ImageRegions/ManageImageRegionsForm.tsx [hunk 3]
```diff
@@ -50,9 +48,11 @@ export const ManageImageRegionsForm = (props: Props) => {
     try {
       await mutateAsync(data);
 
-      enqueueSnackbar('Image regions successfully updated.', {
+      enqueueSnackbar(`${image?.label}'s regions successfully updated.`, {
```
Comment (abailly-akamai) [inline packages/manager/src/features/Images/ImagesLanding/ImageRegions/ManageImageRegionsForm.tsx:51]: You should still provide "Image" as a fallback so we don't display "Undefined" in the snackbar if by any chance we're not getting the image

**Review by abailly-akamai — APPROVED** [review]

**Review by jaalah-akamai — APPROVED** [review]
Nice! Really happy about this change 🥇

File: packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyRegions/AccessKeyRegions.tsx (hunk 2 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/ObjectStorage/AccessKeyLanding/AccessKeyRegions/AccessKeyRegions.tsx]: Ah, fixed with 6df361f

**Pull request merged**